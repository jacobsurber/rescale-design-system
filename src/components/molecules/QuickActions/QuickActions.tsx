import React from 'react';
import { Button, Tooltip } from 'antd';
import { DesktopOutlined, ApartmentOutlined } from '@ant-design/icons';
import styled from 'styled-components';
import { Icon } from '../../atoms/Icon';

export interface QuickAction {
  /** Unique identifier */
  id: string;
  /** Action label */
  label: string;
  /** Action icon */
  icon?: React.ReactNode;
  /** Action description for tooltip */
  description?: string;
  /** Whether action is disabled */
  disabled?: boolean;
  /** Whether action is loading */
  loading?: boolean;
  /** Action type for styling */
  type?: 'primary' | 'secondary' | 'success' | 'warning' | 'danger';
  /** Click handler */
  onClick?: () => void;
}

export interface QuickActionsProps {
  /** Array of quick actions */
  actions?: QuickAction[];
  /** Layout orientation */
  layout?: 'horizontal' | 'vertical' | 'grid';
  /** Button size */
  size?: 'small' | 'default' | 'large';
  /** Whether to show default Rescale actions */
  showDefaults?: boolean;
  /** Whether buttons should be full width in vertical layout */
  fullWidth?: boolean;
  /** Custom className */
  className?: string;
  /** Custom styles */
  style?: React.CSSProperties;
  /** Callback for default actions */
  onNewJob?: () => void;
  onNewWorkstation?: () => void;
  onNewWorkflow?: () => void;
}

const ActionsContainer = styled.div<{ 
  $layout: string; 
  $size: string;
  $fullWidth: boolean;
}>`
  display: ${props => {
    if (props.$layout === 'grid') return 'grid';
    return 'flex';
  }};
  
  ${props => {
    if (props.$layout === 'grid') {
      return `
        grid-template-columns: repeat(auto-fit, minmax(120px, 1fr));
        gap: var(--rescale-space-3);
      `;
    }
    if (props.$layout === 'vertical') {
      return `
        flex-direction: column;
        gap: var(--rescale-space-2);
        ${props.$fullWidth ? 'width: 100%;' : ''}
      `;
    }
    return `
      flex-direction: row;
      gap: var(--rescale-space-3);
      align-items: center;
      flex-wrap: wrap;
    `;
  }}
`;

const ActionButton = styled(Button)<{ 
  $actionType?: string;
  $layout: string;
  $fullWidth: boolean;
}>`
  display: flex;
  align-items: center;
  gap: var(--rescale-space-2);
  
  ${props => props.$layout === 'vertical' && props.$fullWidth && `
    width: 100%;
    justify-content: flex-start;
  `}
  
  ${props => props.$layout === 'grid' && `
    flex-direction: column;
    height: auto;
    padding: var(--rescale-space-4);
    text-align: center;
    gap: var(--rescale-space-2);
  `}
  
  .anticon {
    font-size: ${props => {
      if (props.$layout === 'grid') return '20px';
      return '16px';
    }};
  }
  
  ${props => {
    switch (props.$actionType) {
      case 'success':
        return `
          border-color: var(--rescale-color-success);
          color: var(--rescale-color-success);
          
          &:hover,
          &:focus {
            border-color: var(--rescale-color-success);
            color: var(--rescale-color-success);
            background: var(--rescale-color-success-bg);
          }
        `;
      case 'warning':
        return `
          border-color: var(--rescale-color-warning);
          color: var(--rescale-color-warning);
          
          &:hover,
          &:focus {
            border-color: var(--rescale-color-warning);
            color: var(--rescale-color-warning);
            background: var(--rescale-color-warning-bg);
          }
        `;
      case 'danger':
        return `
          border-color: var(--rescale-color-error);
          color: var(--rescale-color-error);
          
          &:hover,
          &:focus {
            border-color: var(--rescale-color-error);
            color: var(--rescale-color-error);
            background: var(--rescale-color-error-bg);
          }
        `;
      default:
        return '';
    }
  }}
`;

const ActionLabel = styled.span<{ $layout: string }>`
  font-weight: var(--rescale-font-weight-medium);
  
  ${props => props.$layout === 'grid' && `
    font-size: var(--rescale-font-size-sm);
    line-height: 1.4;
  `}
`;

const defaultActions: QuickAction[] = [
  {
    id: 'new-job',
    label: 'New Job',
    icon: <Icon name="PlayCircleOutlined" />,
    description: 'Submit a new simulation job',
    type: 'primary',
  },
  {
    id: 'new-workstation',
    label: 'New Workstation',
    icon: <DesktopOutlined />,
    description: 'Launch a new remote workstation',
    type: 'secondary',
  },
  {
    id: 'new-workflow',
    label: 'New Workflow',
    icon: <ApartmentOutlined />,
    description: 'Create a new automated workflow',
    type: 'secondary',
  },
];

export const QuickActions: React.FC<QuickActionsProps> = ({
  actions = [],
  layout = 'horizontal',
  size = 'default',
  showDefaults = true,
  fullWidth = false,
  className,
  style,
  onNewJob,
  onNewWorkstation,
  onNewWorkflow,
}) => {
  const allActions = showDefaults ? [...defaultActions, ...actions] : actions;

  const handleActionClick = (action: QuickAction) => {
    if (action.disabled || action.loading) return;
    
    // Handle default actions
    if (showDefaults) {
      switch (action.id) {
        case 'new-job':
          onNewJob?.();
          break;
        case 'new-workstation':
          onNewWorkstation?.();
          break;
        case 'new-workflow':
          onNewWorkflow?.();
          break;
      }
    }
    
    // Handle custom action
    action.onClick?.();
  };

  const getButtonType = (action: QuickAction) => {
    switch (action.type) {
      case 'primary':
        return 'primary';
      case 'danger':
        return 'primary';
      default:
        return 'default';
    }
  };

  const renderAction = (action: QuickAction) => {
    const button = (
      <ActionButton
        key={action.id}
        type={getButtonType(action)}
        size={size}
        icon={action.icon}
        loading={action.loading}
        disabled={action.disabled}
        $actionType={action.type}
        $layout={layout}
        $fullWidth={fullWidth}
        onClick={() => handleActionClick(action)}
      >
        <ActionLabel $layout={layout}>
          {action.label}
        </ActionLabel>
      </ActionButton>
    );

    if (action.description) {
      return (
        <Tooltip 
          key={action.id} 
          title={action.description} 
          placement="top"
        >
          {button}
        </Tooltip>
      );
    }

    return button;
  };

  if (allActions.length === 0) {
    return (
      <ActionsContainer
        $layout={layout}
        $size={size}
        $fullWidth={fullWidth}
        className={className}
        style={style}
      >
        <Button
          type="dashed"
          icon={<Icon name="PlusOutlined" />}
          disabled
        >
          No actions available
        </Button>
      </ActionsContainer>
    );
  }

  return (
    <ActionsContainer
      $layout={layout}
      $size={size}
      $fullWidth={fullWidth}
      className={className}
      style={style}
    >
      {allActions.map(renderAction)}
    </ActionsContainer>
  );
};

export default QuickActions;