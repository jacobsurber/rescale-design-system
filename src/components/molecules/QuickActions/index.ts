export { QuickActions } from './QuickActions';
export type { QuickActionsProps, QuickAction } from './QuickActions';