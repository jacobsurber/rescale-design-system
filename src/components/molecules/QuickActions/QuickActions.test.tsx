import { screen, fireEvent, waitFor } from '@testing-library/react';
import { QuickActions } from './QuickActions';
import { 
  render, 
  commonTestScenarios, 
  runAxeTest,
  mockData,
  snapshotHelpers,
} from '../../../test-utils';

describe('QuickActions', () => {
  const mockActions = mockData.generateQuickActions(3);
  const defaultProps = {
    actions: mockActions,
  };

  // Common test scenarios
  commonTestScenarios.shouldRender(QuickActions, defaultProps);
  commonTestScenarios.shouldBeAccessible(QuickActions, defaultProps);
  commonTestScenarios.shouldWorkWithThemes(QuickActions, defaultProps);
  commonTestScenarios.shouldBeResponsive(QuickActions, defaultProps);

  describe('Basic Functionality', () => {
    it('renders all provided actions', () => {
      render(<QuickActions actions={mockActions} />);

      mockActions.forEach(action => {
        expect(screen.getByText(action.title)).toBeInTheDocument();
        if (action.description) {
          expect(screen.getByText(action.description)).toBeInTheDocument();
        }
      });
    });

    it('calls onClick handler when action is clicked', async () => {
      const onActionClick = jest.fn();
      const actionsWithHandler = mockActions.map(action => ({
        ...action,
        onClick: onActionClick,
      }));

      const { user } = render(<QuickActions actions={actionsWithHandler} />);

      const firstAction = screen.getByText(actionsWithHandler[0].title);
      await user.click(firstAction);

      expect(onActionClick).toHaveBeenCalledWith(actionsWithHandler[0]);
    });

    it('handles disabled actions correctly', async () => {
      const onActionClick = jest.fn();
      const disabledAction = {
        ...mockActions[0],
        disabled: true,
        onClick: onActionClick,
      };

      const { user } = render(<QuickActions actions={[disabledAction]} />);

      const actionButton = screen.getByRole('button');
      expect(actionButton).toBeDisabled();

      await user.click(actionButton);
      expect(onActionClick).not.toHaveBeenCalled();
    });

    it('displays loading state correctly', () => {
      const loadingAction = {
        ...mockActions[0],
        loading: true,
      };

      render(<QuickActions actions={[loadingAction]} />);

      const actionButton = screen.getByRole('button');
      expect(actionButton).toHaveClass('ant-btn-loading');
      expect(screen.getByTestId('loading-icon')).toBeInTheDocument();
    });
  });

  describe('Action Display', () => {
    it('shows action icons when provided', () => {
      const actionsWithIcons = mockActions.map(action => ({
        ...action,
        icon: 'plus',
      }));

      render(<QuickActions actions={actionsWithIcons} />);

      const icons = screen.getAllByTestId('action-icon');
      expect(icons).toHaveLength(actionsWithIcons.length);
    });

    it('displays action badges when provided', () => {
      const actionWithBadge = {
        ...mockActions[0],
        badge: { count: 5, color: 'red' },
      };

      render(<QuickActions actions={[actionWithBadge]} />);

      expect(screen.getByText('5')).toBeInTheDocument();
    });

    it('shows tooltips on hover', async () => {
      const actionWithTooltip = {
        ...mockActions[0],
        tooltip: 'This is a helpful tooltip',
      };

      const { user } = render(<QuickActions actions={[actionWithTooltip]} />);

      const actionButton = screen.getByRole('button');
      await user.hover(actionButton);

      await waitFor(() => {
        expect(screen.getByText('This is a helpful tooltip')).toBeInTheDocument();
      });
    });
  });

  describe('Layout and Styling', () => {
    it('applies different sizes correctly', () => {
      const sizes = ['small', 'default', 'large'] as const;
      
      sizes.forEach(size => {
        const { container, unmount } = render(
          <QuickActions actions={mockActions} size={size} />
        );

        const actionsContainer = container.querySelector('.quick-actions');
        expect(actionsContainer).toHaveClass(`quick-actions-${size}`);
        unmount();
      });
    });

    it('applies vertical layout when specified', () => {
      const { container } = render(
        <QuickActions actions={mockActions} layout="vertical" />
      );

      const actionsContainer = container.querySelector('.quick-actions');
      expect(actionsContainer).toHaveClass('quick-actions-vertical');
    });

    it('applies horizontal layout by default', () => {
      const { container } = render(<QuickActions actions={mockActions} />);

      const actionsContainer = container.querySelector('.quick-actions');
      expect(actionsContainer).toHaveClass('quick-actions-horizontal');
    });

    it('handles different action types with appropriate styling', () => {
      const actionTypes = ['primary', 'secondary', 'danger'] as const;
      const typedActions = mockActions.slice(0, 3).map((action, index) => ({
        ...action,
        type: actionTypes[index],
      }));

      render(<QuickActions actions={typedActions} />);

      const buttons = screen.getAllByRole('button');
      expect(buttons[0]).toHaveClass('ant-btn-primary');
      expect(buttons[1]).toHaveClass('ant-btn-default');
      expect(buttons[2]).toHaveClass('ant-btn-dangerous');
    });
  });

  describe('Advanced Features', () => {
    it('supports action groups with separators', () => {
      const groupedActions = [
        { ...mockActions[0], group: 'create' },
        { ...mockActions[1], group: 'create' },
        { ...mockActions[2], group: 'manage' },
      ];

      const { container } = render(<QuickActions actions={groupedActions} showGroups />);

      const separators = container.querySelectorAll('.action-group-separator');
      expect(separators.length).toBe(1); // One separator between groups
    });

    it('displays keyboard shortcuts when provided', () => {
      const actionWithShortcut = {
        ...mockActions[0],
        shortcut: 'Ctrl+N',
      };

      render(<QuickActions actions={[actionWithShortcut]} showShortcuts />);

      expect(screen.getByText('Ctrl+N')).toBeInTheDocument();
    });

    it('handles action overflow with dropdown menu', async () => {
      const manyActions = mockData.generateQuickActions(10);
      const { user } = render(
        <QuickActions actions={manyActions} maxVisible={5} />
      );

      // Should show 5 actions plus overflow button
      const visibleButtons = screen.getAllByRole('button');
      expect(visibleButtons.length).toBe(6); // 5 + overflow

      const overflowButton = screen.getByTestId('overflow-actions');
      await user.click(overflowButton);

      // Should show remaining actions in dropdown
      await waitFor(() => {
        const overflowActions = manyActions.slice(5);
        overflowActions.forEach(action => {
          expect(screen.getByText(action.title)).toBeInTheDocument();
        });
      });
    });
  });

  describe('Interactive States', () => {
    it('provides visual feedback on hover', async () => {
      const { user } = render(<QuickActions actions={mockActions} />);

      const firstButton = screen.getByText(mockActions[0].title);
      await user.hover(firstButton);

      expect(firstButton.closest('button')).toHaveClass('ant-btn-hover');
    });

    it('shows focus indicators for keyboard navigation', async () => {
      const { user } = render(<QuickActions actions={mockActions} />);

      const firstButton = screen.getAllByRole('button')[0];
      firstButton.focus();

      expect(firstButton).toHaveFocus();
      expect(firstButton).toHaveClass('ant-btn-focused');
    });

    it('supports keyboard activation with Enter and Space', async () => {
      const onActionClick = jest.fn();
      const actionWithHandler = {
        ...mockActions[0],
        onClick: onActionClick,
      };

      const { user } = render(<QuickActions actions={[actionWithHandler]} />);

      const button = screen.getByRole('button');
      button.focus();

      await user.keyboard('{Enter}');
      expect(onActionClick).toHaveBeenCalledTimes(1);

      await user.keyboard(' ');
      expect(onActionClick).toHaveBeenCalledTimes(2);
    });
  });

  describe('Error Handling', () => {
    it('handles empty actions array gracefully', () => {
      const { container } = render(<QuickActions actions={[]} />);
      
      expect(container.firstChild).toBeInTheDocument();
      expect(screen.queryByRole('button')).not.toBeInTheDocument();
    });

    it('handles actions with missing required props', () => {
      const incompleteAction = {
        id: 'incomplete',
        // Missing title
      } as any;

      // Should not crash, might show fallback or skip the action
      const { container } = render(<QuickActions actions={[incompleteAction]} />);
      expect(container.firstChild).toBeInTheDocument();
    });

    it('handles onClick errors gracefully', async () => {
      const errorAction = {
        ...mockActions[0],
        onClick: () => {
          throw new Error('Test error');
        },
      };

      const consoleSpy = jest.spyOn(console, 'error').mockImplementation();
      const { user } = render(<QuickActions actions={[errorAction]} />);

      const button = screen.getByRole('button');
      await user.click(button);

      expect(consoleSpy).toHaveBeenCalled();
      consoleSpy.mockRestore();
    });
  });

  describe('Performance', () => {
    it('memoizes action buttons to prevent unnecessary re-renders', () => {
      const { rerender } = render(<QuickActions actions={mockActions} />);

      // Re-render with same actions
      rerender(<QuickActions actions={mockActions} />);

      // Actions should remain stable (this would be tested with React DevTools Profiler in practice)
      expect(screen.getAllByRole('button')).toHaveLength(mockActions.length);
    });

    it('handles rapid clicks without double-execution', async () => {
      const onActionClick = jest.fn();
      const actionWithHandler = {
        ...mockActions[0],
        onClick: onActionClick,
      };

      const { user } = render(<QuickActions actions={[actionWithHandler]} />);

      const button = screen.getByRole('button');
      
      // Rapid clicks
      await user.click(button);
      await user.click(button);
      await user.click(button);

      // Should handle debouncing or prevent double execution
      expect(onActionClick).toHaveBeenCalledTimes(3);
    });
  });

  describe('Accessibility', () => {
    it('provides proper ARIA labels for actions', () => {
      const actionWithLabel = {
        ...mockActions[0],
        ariaLabel: 'Create new job action',
      };

      render(<QuickActions actions={[actionWithLabel]} />);

      const button = screen.getByRole('button');
      expect(button).toHaveAccessibleName('Create new job action');
    });

    it('groups related actions with proper ARIA attributes', () => {
      const groupedActions = mockActions.map(action => ({
        ...action,
        group: 'creation-actions',
      }));

      const { container } = render(
        <QuickActions actions={groupedActions} showGroups />
      );

      const group = container.querySelector('[role="group"]');
      expect(group).toBeInTheDocument();
      expect(group).toHaveAttribute('aria-label');
    });

    it('announces state changes to screen readers', async () => {
      const actionWithState = {
        ...mockActions[0],
        loading: false,
      };

      const { rerender } = render(<QuickActions actions={[actionWithState]} />);

      // Change to loading state
      rerender(
        <QuickActions 
          actions={[{ ...actionWithState, loading: true }]} 
        />
      );

      const button = screen.getByRole('button');
      expect(button).toHaveAttribute('aria-busy', 'true');
    });

    it('passes accessibility audit', async () => {
      const { container } = render(
        <QuickActions 
          actions={mockActions}
          showShortcuts
          showGroups
        />
      );

      await runAxeTest(container);
    });
  });

  describe('Snapshots', () => {
    it('matches snapshot with basic configuration', () => {
      snapshotHelpers.createComponentSnapshot(QuickActions, {
        actions: mockActions.slice(0, 3),
      });
    });

    it('matches snapshot with advanced features', () => {
      const advancedActions = mockActions.map(action => ({
        ...action,
        icon: 'plus',
        badge: { count: 2, color: 'blue' },
        tooltip: `Tooltip for ${action.title}`,
        shortcut: 'Ctrl+N',
      }));

      snapshotHelpers.createComponentSnapshot(QuickActions, {
        actions: advancedActions,
        layout: 'vertical',
        size: 'large',
        showShortcuts: true,
        showGroups: true,
      });
    });
  });
});