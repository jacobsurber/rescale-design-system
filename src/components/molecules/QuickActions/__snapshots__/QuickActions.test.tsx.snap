// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`QuickActions Snapshots matches snapshot with advanced features 1`] = `
<div
  class="sc-bRKDuR jKddit"
>
  <button
    aria-describedby="test-id"
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-primary ant-btn-color-primary ant-btn-variant-solid ant-btn-lg sc-hvigdm bWQmCz"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <span
        aria-label="play-circle"
        class="anticon anticon-play-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="play-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
          <path
            d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
          />
        </svg>
      </span>
    </span>
    <span
      class="sc-fhHczv czouPK"
    >
      New Job
    </span>
  </button>
  <button
    aria-describedby="test-id"
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-default ant-btn-color-default ant-btn-variant-outlined ant-btn-lg sc-hvigdm bWQmCz"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <span
        aria-label="desktop"
        class="anticon anticon-desktop"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="desktop"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M928 140H96c-17.7 0-32 14.3-32 32v496c0 17.7 14.3 32 32 32h380v112H304c-8.8 0-16 7.2-16 16v48c0 4.4 3.6 8 8 8h432c4.4 0 8-3.6 8-8v-48c0-8.8-7.2-16-16-16H548V700h380c17.7 0 32-14.3 32-32V172c0-17.7-14.3-32-32-32zm-40 488H136V212h752v416z"
          />
        </svg>
      </span>
    </span>
    <span
      class="sc-fhHczv czouPK"
    >
      New Workstation
    </span>
  </button>
  <button
    aria-describedby="test-id"
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-default ant-btn-color-default ant-btn-variant-outlined ant-btn-lg sc-hvigdm bWQmCz"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <span
        aria-label="apartment"
        class="anticon anticon-apartment"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="apartment"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M908 640H804V488c0-4.4-3.6-8-8-8H548v-96h108c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16H368c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h108v96H228c-4.4 0-8 3.6-8 8v152H116c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16H292v-88h440v88H620c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16zm-564 76v168H176V716h168zm84-408V140h168v168H428zm420 576H680V716h168v168z"
          />
        </svg>
      </span>
    </span>
    <span
      class="sc-fhHczv czouPK"
    >
      New Workflow
    </span>
  </button>
  <button
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-primary ant-btn-color-primary ant-btn-variant-solid ant-btn-lg sc-hvigdm bWQmCz"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      plus
    </span>
    <span
      class="sc-fhHczv czouPK"
    >
      Action 1
    </span>
  </button>
  <button
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-default ant-btn-color-default ant-btn-variant-outlined ant-btn-lg sc-hvigdm bWQmCz"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      plus
    </span>
    <span
      class="sc-fhHczv czouPK"
    >
      Action 2
    </span>
  </button>
  <button
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-default ant-btn-color-default ant-btn-variant-outlined ant-btn-lg sc-hvigdm ctCtkJ"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      plus
    </span>
    <span
      class="sc-fhHczv czouPK"
    >
      Action 3
    </span>
  </button>
</div>
`;

exports[`QuickActions Snapshots matches snapshot with basic configuration 1`] = `
<div
  class="sc-bRKDuR hAXCNx"
>
  <button
    aria-describedby="test-id"
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-primary ant-btn-color-primary ant-btn-variant-solid sc-hvigdm bWQmCz"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <span
        aria-label="play-circle"
        class="anticon anticon-play-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="play-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
          <path
            d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
          />
        </svg>
      </span>
    </span>
    <span
      class="sc-fhHczv czouPK"
    >
      New Job
    </span>
  </button>
  <button
    aria-describedby="test-id"
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-default ant-btn-color-default ant-btn-variant-outlined sc-hvigdm bWQmCz"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <span
        aria-label="desktop"
        class="anticon anticon-desktop"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="desktop"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M928 140H96c-17.7 0-32 14.3-32 32v496c0 17.7 14.3 32 32 32h380v112H304c-8.8 0-16 7.2-16 16v48c0 4.4 3.6 8 8 8h432c4.4 0 8-3.6 8-8v-48c0-8.8-7.2-16-16-16H548V700h380c17.7 0 32-14.3 32-32V172c0-17.7-14.3-32-32-32zm-40 488H136V212h752v416z"
          />
        </svg>
      </span>
    </span>
    <span
      class="sc-fhHczv czouPK"
    >
      New Workstation
    </span>
  </button>
  <button
    aria-describedby="test-id"
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-default ant-btn-color-default ant-btn-variant-outlined sc-hvigdm bWQmCz"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <span
        aria-label="apartment"
        class="anticon anticon-apartment"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="apartment"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M908 640H804V488c0-4.4-3.6-8-8-8H548v-96h108c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16H368c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h108v96H228c-4.4 0-8 3.6-8 8v152H116c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16H292v-88h440v88H620c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16zm-564 76v168H176V716h168zm84-408V140h168v168H428zm420 576H680V716h168v168z"
          />
        </svg>
      </span>
    </span>
    <span
      class="sc-fhHczv czouPK"
    >
      New Workflow
    </span>
  </button>
  <button
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-primary ant-btn-color-primary ant-btn-variant-solid sc-hvigdm bWQmCz"
    type="button"
  >
    <span
      class="sc-fhHczv czouPK"
    >
      Action 1
    </span>
  </button>
  <button
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-default ant-btn-color-default ant-btn-variant-outlined sc-hvigdm bWQmCz"
    type="button"
  >
    <span
      class="sc-fhHczv czouPK"
    >
      Action 2
    </span>
  </button>
  <button
    class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-default ant-btn-color-default ant-btn-variant-outlined sc-hvigdm ctCtkJ"
    type="button"
  >
    <span
      class="sc-fhHczv czouPK"
    >
      Action 3
    </span>
  </button>
</div>
`;
